### PR TITLE
Add a setting for navigation background

### LINKED ISSUE #1354: Create the navigation settings
I am finding that we need to manipulate the background colour of the navigation in many places. I think we should simplify the navigation pattern by removing the dark modification and adding a `$color-navigation-background` variable. This can be white or brand by default.

The navigation should detect if the colour is dark or light and change the font colour accordingly.

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Add a setting for navigation background color

## QA

- Pull code
- Run `./run serve --watch`
- Open and check nothing broken on:
http://0.0.0.0:8101/vanilla-framework/examples/patterns/navigation/default/
http://0.0.0.0:8101/vanilla-framework/examples/patterns/navigation/dark/
http://0.0.0.0:8101/vanilla-framework/examples/patterns/navigation/light/

## Details

Fixes #1354

### CODE REVIEW

File: scss/_patterns_navigation.scss [hunk 1]
```diff
@@ -3,35 +3,17 @@
...
-      $background-color: $color-dark,
-      $border-color: $color-mid-light,
-      $text-color: $color-x-light
-    );
-  }
-}
-
-@mixin navigation-light {
-  .p-navigation--light {
+@mixin navigation-default {
```
Comment (anthonydillon) [inline scss/_patterns_navigation.scss]: Do we need default? When we have a single pattern don't we just put the style directly in vf mixin.

File: examples/patterns/navigation/default.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
 category: _patterns
 ---
 
-<header id="navigation" class="p-navigation--light">
+<header id="navigation" class="p-navigation">
```
Comment (anthonydillon) [inline examples/patterns/navigation/default.html]: If this is the only example of the pattern please remove the navigation directory and rename this file `navigation.md`

File: docs/en/settings/color-settings.md [hunk 3]
```diff
@@ -76,6 +76,10 @@ When planning your pages, make sure there is an even distribution and well-balan
 <td>$color-accent-background</td>
 <td style="background-color: #333; color: #fff;">$color-brand</td>
 </tr>
+<tr>
+<td>$color-accent-background</td>
+<td style="background-color: #fff; color: #000;">$color-x-light</td>
```
Comment (anthonydillon) [inline docs/en/settings/color-settings.md]: This variable is set to `$accent-color` which in turn is set to `#333`

File: scss/_patterns_navigation.scss (hunk 1 above)
Comment (barrymcgee) [reply scss/_patterns_navigation.scss]: There's two patterns - default and sidebar?

File: examples/patterns/navigation/default.html (hunk 2 above)
Comment (barrymcgee) [reply examples/patterns/navigation/default.html]: As above, two patterns - default and sidebar?

File: scss/_patterns_navigation.scss (hunk 1 above)
Comment (anthonydillon) [reply scss/_patterns_navigation.scss]: Aw, good point. Sorry, didn't see the sidebar in the diff.

**Comment by anthonydillon** [comment]
CI is picking up on the variable name. If you put it in back ticks the spell checker ignores it.

**Pull request merged**